store/cachekv/store.go

```diff
@@ -32,6 +32,12 @@ type Store struct {
 	parent        types.KVStore
 }
 
+// PooledStore wraps a Store object and implements the types.PooledCacheKVStore interface,
+// which allows it to be pooled and reused without the overhead of allocation.
+type PooledStore struct {
```

Comment:
we should add an interface assertion above this struct declaration ala `var _ types.CacheKVStore = (*Store)(nil)`